mmv1/products/clouddeploy/terraform.yaml

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Overrides::ResourceOverrides
+  DeliveryPipeline: !ruby/object:Overrides::Terraform::ResourceOverride
+    autogen_async: true
+    id_format: '{{name}}'
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "deploy_delivery_pipeline"
+        primary_resource_id: "pipeline"
+        vars:
+          name: "dev"
+    properties:
+      name: !ruby/object:Overrides::Terraform::PropertyOverride
+        custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
+      region: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+  Target: !ruby/object:Overrides::Terraform::ResourceOverride
+    autogen_async: true
+    id_format: '{{name}}'
```

Comment:
Is `name` the right format? This is the short-form, so won't be able to identify the full resource during import?

Reply:
Yes 

https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5710-old..auto-pr-5710#diff-e36b98e511d7d496a126e50204560eea400fe43213204672a22acece89f3f4ebR409

Reply:
On a side note, I'm not a fan of something_Id as name is the consistent resource name in most resources. The API uses name as selflink but an identical value can be read with the id attribute.

Reply:
So, having this id_format ends up with the following code to set the ID of the resource:
```
	// Store the ID now
	id, err := replaceVars(d, config, "{{name}}")
	if err != nil {
		return fmt.Errorf("Error constructing id: %s", err)
	}
	d.SetId(id)
```

I don't believe this is right, because it doesn't represent the full resource. The ID format should be `projects/{{project}}/locations/{{region}}/deliveryPipelines/{{name}}` so that we can reference this value from other resources within a config (if needed)